libcxx/include/deque

```diff
@@ -2543,7 +2543,8 @@ inline void deque<_Tp, _Allocator>::clear() _NOEXCEPT {
 }
 
 template <class _Tp, class _Allocator>
-inline _LIBCPP_HIDE_FROM_ABI bool operator==(const deque<_Tp, _Allocator>& __x, const deque<_Tp, _Allocator>& __y) {
+inline _LIBCPP_CONSTEXPR_SINCE_CXX26 constexpr bool
+operator==(const deque<_Tp, _Allocator>& __x, const deque<_Tp, _Allocator>& __y) {
```

Comment:
`_LIBCPP_HIDE_FROM_ABI` should be kept. Also, it's incorrect to duplicate `constexpr`. Ditto below.

Member functions of `deque` also need to be changed. And we need to carefully handle code snippets for ASAN.

```suggestion
_LIBCPP_HIDE_FROM_ABI inline _LIBCPP_CONSTEXPR_SINCE_CXX26 bool
operator==(const deque<_Tp, _Allocator>& __x, const deque<_Tp, _Allocator>& __y) {
```